### PR TITLE
Itemset products

### PR DESCRIPTION
**Pull request by emord**
@czue 

Support for products in itemsets. I threw it in with fixtures since they are both represented as fixtures in the phone restore. Could possibly break it out into it's own endpoint once vellum's data source module is more generic, but probably not a need to.

@esoergel I went ahead and supported custom data, although it does look like the phone restore XML is not homogenous when using that.

### CODE REVIEW

File: corehq/apps/custom_data_fields/models.py [hunk 1]
```diff
@@ -54,10 +54,8 @@ def _is_match(field):
         return filter(_is_match, self.fields)
 
     @classmethod
-    def get_or_create(cls, domain, field_type):
-        # todo: this overrides get_or_create from DocumentBase but with a completely different signature.
-        # This method should probably be renamed.
-        existing = cls.view(
+    def get_by_domain_and_type(cls, domain, field_type):
+        return cls.view(
```
Comment (czue) [inline corehq/apps/custom_data_fields/models.py]: while you were in here maybe take the opportunity to move to the `dbaccessors` module?

File: corehq/apps/products/fixtures.py [hunk 2]
```diff
@@ -1,17 +1,47 @@
...
+    'program_id',
+    'cost',
+    'product_data'
+]
+
+CUSTOM_DATA_SLUG = 'product_data'
+
+
+def product_fixture_generator_json(domain):
+    if SQLProduct.objects.filter(domain=domain).count() == 0:
```
Comment (czue) [inline corehq/apps/products/fixtures.py]: might want to use `.exists()` here. not 100% true whether it matters for performance, but even for readability is a small improvement

**Comment by czue** [comment]
thanks for turning this around so quickly! a couple small comments, but nothing blocking.

File: corehq/apps/custom_data_fields/models.py (hunk 1 above)
Comment (esoergel) [inline corehq/apps/custom_data_fields/models.py]: wouldn't hurt to address that TODO if it looks easy too.

File: corehq/apps/products/fixtures.py [hunk 3]
```diff
@@ -1,17 +1,47 @@
...
+            fields.append(CUSTOM_DATA_SLUG + '/' + f.slug)
+
+    return {
+        'sourceUri': 'jr://fixture/commtrack:products',
+        'defaultId': 'products',
+        'initialQuery': "instance('products')/products/product",
+        'name': 'Products',
+        'structure': {
+            f: {
+                'name': f,
```
Comment (millerdev) [inline corehq/apps/products/fixtures.py]: What's this for? The [documentation](https://github.com/dimagi/Vellum/blob/master/src/datasources.js) doesn't mention it.

Comment (emord) [inline corehq/apps/products/fixtures.py]: Fixed: https://github.com/dimagi/Vellum/commit/693cffa618830d1aab650a8cccff644dd0d9f6f4

**Pull request merged**